src/libsync/progressdispatcher.cpp

```diff
@@ -195,6 +195,27 @@ void ProgressInfo::adjustTotalsForFile(const SyncFileItem &item)
     }
 }
 
+void ProgressInfo::updateTotalsForFile(const SyncFileItem &item, quint64 newSize)
+{
+    if (!shouldCountProgress(item)) {
+        return;
+    }
+
+    // Update item's progress total
+    setProgressItem(item, 0);
+    _currentItems[item._file]._progress._total = newSize;
+
+    // Update _sizeProgress._total
+    _sizeProgress._total = 0;
+    QHashIterator<QString, ProgressItem> it(_currentItems);
+    while (it.hasNext()) {
+        it.next();
+        if (isSizeDependent(it.value()._item)) {
+            _sizeProgress._total += it.value()._progress._total;
```

Comment:
I don't think this is correct. As far as I'm aware `_currentItems` always only has a subset of the items -- the ones that are currently in progress. Items that haven't started yet and items that have finished aren't in there. That means this sum will usually be way less than the actual total size.

I expect you want to make this function safe to call multiple times for the same item (I think that's a good idea). Suggestion:
```
if (!_currentItems.contains(item._file)) {
    _sizeProgress._total += newSize - item._size
} else {
    _sizeProgress._total += newSize - _currentItems[item._file]._progress._total;
}

// now setup _currentItems[item._file]
```

Reply:
That's exactly what I was trying to do! Thanks for the explanation of what `progressdispatcher` is actually doing 😄

Reply:
When you say `no setup _currentItems[item._file]`, what exactly needs to be done? I thought the code above it was enough?

Reply:
I was adding it to point out that you can't do `setProgressItem(item, 0);` before, it has to come after.

Reply:
You must ensure that `_currentItems[item._file]._progress._total == newSize` after this function is done, otherwise a second call might do nonsense.

Reply:
With an assert? enforce? or is there a smart way to handle if somehow they get out-of-sync?

Reply:
No, it's just that https://github.com/owncloud/client/pull/6131/commits/ee7bc2da1916faffe0df6c8ad374b44964fe7789 doesn't currently seem to set _currentItems[item._file] at all, but it should!

Reply:
Yeah I realized after I wrote that what you meant, done.